design-guidelines/interactive-element-states.md

```diff
@@ -11,7 +11,33 @@ States communicate the different statuses of UI components to the user, includin
 - Data inputs
   - For example: a text input
 
-## States
+## States: Ready to receive input
+
+### Hover
+
+Styling a component when it is hovered by a cursor indicates that a component can be interacted with. If a component is normally interactive but is currently disabled, it should not be styled on hover. Do not remove the browser's default hover styles unless you're overriding them with custom hover styles.
+
+In addition to updating a component's style when it is hovered, it is also a good idea to update the cursor style to match the purpose of the component. For example: showing a resize cursor when hovering a drag handle that is used to resize a table column.
+
+### Focus
+
+Styling a component when it receives focus indicates that a component is active and ready to receive input. If a component is normally interactive but is currently disabled, it should not receive focus. Do not remove the browser's default focus styles unless you're overriding them with custom focus styles.
+
+Focus styles can match the hover styles, but it's also okay to have different styles to indicate hover and focus. It's also acceptable to have different focus styles for different input devices. For example: our button components have focus styles that only appear when the button is focused using the keyboard.
+
+### Active
+
+A component can be styled to indicate it is actively receiving input such as a mouse click. If a component is normally interactive but is currently disabled, it should not receive active styles. Do not remove the browser's default active styles unless you're overriding them with custom active styles.
+
+### Selected
+
+If a component has been toggled to an "on" state, or picked from a list of options, it should be styled to visually indicate its selected state. If there is another affordance to indicate the selected state, such as a checkmark or a text label, additional styles are optional.
+
+## States: Not ready to receive input
+
+### Loading
+
+If an action will not be completed immediately, the user should be reassured that their action has not failed but it is still in progress. The component that triggers the action should be disabled to avoid re-starting the process. If the action has a label, the label may be updated to communicate what's happening. For example: changing a button's label from "Save" to "Saving". Loading states should not block interaction with unrelated components.
```

Comment:
You're right, not all labels need to be updated. That's why I said

> the label _may be_ updated to communicate what's happening

Reply:
> shouldn't we put page loading and card loading here as a case here? for case where loading indicates that content is coming.

Page loading and card loading will be addressed by https://jira.d2iq.com/browse/D2IQ-69562

These guidelines refer only to interactive elements like buttons and inputs.

Reply:
> I think we should give specific examples here like you did with save -> saving

Why do we need other examples? Is something unclear about the "Save => Saving..." example?

> and consider the progress animated icon as well.

None of our interactive components have an animated icon yet

Reply:
https://github.com/dcos-labs/ui-kit/pull/536#discussion_r474228981

OK but what about selection-depedent / progressive disclosure? 
I will put a note in the jira that good to cross reference between these documents when other is live as i believe will help readers find right informaiton.

Reply:
> > I think we should give specific examples here like you did with save -> saving
> 
> Why do we need other examples? Is something unclear about the "Save => Saving..." example?
> 
> > and consider the progress animated icon as well.
> 
> None of our interactive components have an animated icon yet

it depends if you feel selection-dependent / progressive disclosure should be here?
also, should we address statuses like (loading icon) Processing..." as "loading" ?

For button text standards for this case, I think we should include the ... for the Saving text ? I think it will help with readability and recognition that this is loading. do you think we should also include the processing animated icon? 

As I am writing this, do you think we should distinguish more between loading and processing? I think it would be helpful for the standards and simplicity so don't overload loading especially as you have typical loading aspects in another guideline. especially as "If an action will not be completed immediately, the user should be reassured that their action has not failed but it is still in progress." envelopes the case of card and page loading

Reply:
> OK but what about selection-depedent / progressive disclosure?

I wouldn't consider those to be relevant to guidelines about a loading/processing state.

> For button text standards for this case, I think we should include the ... for the Saving text

Ok, I'll add that

> As I am writing this, do you think we should distinguish more between loading and processing?

Possibly? What do you think the difference between "loading" and "processing" is?